### PR TITLE
Rename: Check for existence of BufferUndoState in UndoTemporaryEdits

### PR DESCRIPTION
**Pull request by dpoeschl**
Fixes part of internal TFS bug #1167415.

We have reports of the BufferUndoState being unavailable in
UndoTemporaryEdits when inline rename has been dismissed due to an
external workspace change. This change prevents the crash by checking
for the existence of the relevant BufferUndoState while we await more
information on what leads to this state.

Tagging for review: @Pilchie @jasonmalinowski @rchande @balajikris @brettfo @jmarolf

### CODE REVIEW

**Comment by jmarolf** [comment]
:+1:

File: src/VisualStudio/Core/Def/Implementation/InlineRename/InlineRenameUndoManager.cs [hunk 1]
```diff
@@ -144,9 +144,17 @@ public void UndoTemporaryEdits(ITextBuffer subjectBuffer, bool disconnect)
 
             protected override void UndoTemporaryEdits(ITextBuffer subjectBuffer, bool disconnect, bool undoConflictResolution)
             {
-                var undoManager = this.UndoManagers[subjectBuffer].UndoManager;
-                var startRenameUndoPrimitive = this.UndoManagers[subjectBuffer].StartRenameSessionUndoPrimitive;
-                var markerPrimitive = this.UndoManagers[subjectBuffer].ConflictResolutionRenameUndoPrimitive ?? startRenameUndoPrimitive;
+                // We have reports of the BufferUndoState being unavailable here when inline rename
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/InlineRename/InlineRenameUndoManager.cs]: "we have reports" -- do we have an internal repro of this? Just making sure the comment is accurate for whoever will next get the honor of seeing this.

**Comment by dpoeschl** [comment]
@jasonmalinowski Comment updated.

**Comment by dpoeschl** [comment]
Received verbal signoff from @jasonmalinowski

**Pull request merged**